app/api/routes.py

```diff
@@ -465,31 +486,64 @@ def update_resource(id, json, db):
         return utils.standardize_response(status_code=500)
 
 
-def create_resource(json, db):
-    langs, categ = get_attributes(json)
-    new_resource = Resource(
-        name=json.get('name'),
-        url=json.get('url'),
-        category=categ,
-        languages=langs,
-        paid=json.get('paid'),
-        notes=json.get('notes'))
-
+def create_resources(json, db):
     try:
-        db.session.add(new_resource)
-        db.session.commit()
-        index.save_object(new_resource.serialize_algolia_search)
-
-    except (AlgoliaUnreachableHostException, AlgoliaException) as e:
-        logger.exception(e)
-        print(f"Algolia failed to index new resource '{new_resource.name}'")
-
-    except IntegrityError as e:
-        logger.exception(e)
-        return utils.standardize_response(status_code=422)
+        # Resources to return in the response
+        created_resources = []
+        # Serialized Resources to send to Algolia
+        created_resources_algolia = []
+        # Resource IDs to delete if Algolia fails to index
+        resource_id_cache = []
+
+        # Create each Resource in the database one by one
+        for resource in json:
+            langs, categ = get_attributes(resource)
+            new_resource = Resource(
+                name=resource.get('name'),
+                url=resource.get('url'),
+                category=categ,
+                languages=langs,
+                paid=resource.get('paid'),
+                notes=resource.get('notes'))
+
+            try:
+                db.session.add(new_resource)
+                db.session.commit()
+                created_resources_algolia.append(new_resource.serialize_algolia_search)
+                resource_id_cache.append(new_resource.id)
+
+            except IntegrityError as e:
+                logger.exception(e)
+                return utils.standardize_response(status_code=422)
+
+            except Exception as e:
+                logger.exception(e)
+                return utils.standardize_response(status_code=500)
+
+            created_resources.append(new_resource.serialize)
+
+        # Take all the created resources and save them in Algolia with one API call
+        try:
+            index.save_objects(created_resources_algolia)
 
+        except (AlgoliaUnreachableHostException, AlgoliaException) as e:
+            if (environ.get("FLASK_ENV") != 'development'):
+                logger.exception(e)
+
+                # Remove created resources from the db to stay in sync with Algolia
+                for res_id in resource_id_cache:
+                    res = Resource.query.get(res_id)
+                    res.languages.clear()
+                    res = Resource.query.filter_by(id=res_id)
+                    res.delete()
```

Comment:
I know this looks really funky, it's due to my lack of knowledge about flask-sqlalchemy. This was the only way I could successfully get the resource to delete because of the reference to the resource in the `language_identifiers` table. I'm not sure what the difference is between the object returned from `Resource.query.get(res_id)` and the one returned from `Resource.query.filter_by(id=res_id)`.

If someone knows how to delete a resource in a one liner, please comment and I'll update it.